ray-operator/controllers/ray/common/pod.go

```diff
@@ -89,16 +89,12 @@ func initTemplateAnnotations(instance rayv1.RayCluster, podTemplate *corev1.PodT
 
 // DefaultHeadPodTemplate sets the config values
 func DefaultHeadPodTemplate(ctx context.Context, instance rayv1.RayCluster, headSpec rayv1.HeadGroupSpec, podName string, headPort string) corev1.PodTemplateSpec {
-	log := ctrl.LoggerFrom(ctx)
 	// TODO (Dmitri) The argument headPort is essentially unused;
 	// headPort is passed into setMissingRayStartParams but unused there for the head pod.
 	// To mitigate this awkwardness and reduce code redundancy, unify head and worker pod configuration logic.
 	podTemplate := headSpec.Template
 	podTemplate.GenerateName = podName
-	if podTemplate.ObjectMeta.Namespace == "" {
-		podTemplate.ObjectMeta.Namespace = instance.Namespace
-		log.Info("Setting pod namespaces", "namespace", instance.Namespace)
-	}
+	podTemplate.ObjectMeta.Namespace = instance.Namespace
```

Comment:
Maybe its better to return an error in this case overriding user values seems wrong?